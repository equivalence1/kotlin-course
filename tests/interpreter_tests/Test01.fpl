fun sum(a, b) {
    return a + b
}

println(sum(1, 2))

var a = 10
var b = 20
if (a > b) {
    println(1)
} else {
    println(0)
}

fun fib(n) {
    if (n <= 1) {
        return 1
    }
    return fib(n - 1) + fib(n - 2)
}

var i = 1
while (i <= 5) {
    println(i, fib(i))
    i = i + 1
}

fun foo(n) {
    fun bar(m) {
        return m + n
    }

    return bar(1)
}

println(foo(41))

b = a + (a + a) * a % a / a     // b = 10
println(a)
println(b)
println(b < a)
println(b > a)
println(b || a)
println(b <= a)
println(b >= a)
println(b && a)
println(b == a)
println(b != a)
println(0 || 1)
println(1 && 0)

if (a || b || 1) {
    fun f(n, m) {
    }
    f(a, b)
} else {
    while (a) {
        while(b) {
            println(123)
        }
    }
}