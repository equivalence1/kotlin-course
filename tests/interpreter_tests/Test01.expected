3
0
1, 1
2, 2
3, 3
4, 5
5, 8
42
10
10
0
0
1
1
1
1
1
0
1
0